core/src/main/java/io/questdb/griffin/ExpressionParser.java

```diff
@@ -1003,6 +1003,8 @@ void parseExpr(
                                 throw SqlException.$(lastPos, "constant is not allowed here");
                             }
                             break;
+                        } else if (SqlKeywords.isForKeyword(tok)) {
```

Comment:
lets not use hacks to exit expression parser like this. Expression parser will exit on standalone token automatically, e.g. `a + b c`, parser exits cleanly when it sees "c"